### ISSUE TITLE
Extract URL embedded in path

### ISSUE THREAD

**Issue by plicit** [TrianguloY/UrlChecker#169]
**Is your feature request related to a problem? Please describe.**

Some services embed the redirect URL in the path rather than in the query string, for example (non-working, anonymized):

https://tracker.example.com/redirect/https:%2F%2Fdestination.com%2Fdest-path%2F%3Futm_source=redirector/tracking-codes

**Describe the solution you'd like**

I'd like URL Check to scan for something like /https?:%2F.*?(/|\?|$) and extract the URL (delimited by / or end of path component).

**Describe alternatives you've considered**

I like that URL Check lets me edit the URL directly but it's difficult in this case because the embedded link is URL encoded.

**Comment by TrianguloY** [comment]
Btw, I know it's probably intricate and hard to use, but if you want a workaround I think you can (haven't tested):
- create a pattern in the patterns checker module to detect the urls (like yours) and use a replacement that will set it as "?url=$1", for example "?url=http:%2Fsomething"
- expand the query remover list and click on the parsed query. 

The query remover does decode uris, so it can be used as a decoder with this setup. Maybe doesn't work as expected, but could be fun to try if someone wants to test.

**Comment by plicit** [comment]
That is a brilliant idea, thanks!  I didn't realize each module allowed you to edit its rules.

I first tried the pattern checker but it looks like the URL cleaner does not execute again after the pattern checker did the replacement.

However, I then edited the URL cleaner rules, and this worked perfectly:
```json
    "URL in path": {
      "urlPattern": "^(?:.+?)\\\/(https?:%2F.+?)(\\\/|\\?|$).*",
      "redirections": [
        "^(?:.+?)\\\/(https?:%2F.+?)(\\\/|\\?|$).*"
      ]
    },
```

Thank you for creating this wonderful app! 🙂
